### PR TITLE
maint: change license to Apache v2 from MIT

### PR DESCRIPTION
**Pull request by dbelyaev**
update default license to **Apache v2** from **MIT** to avoid breaking Google license agreement - they are referring **CC-By 3.0** in their style guide (ref: https://google.github.io/styleguide/) while using **Apache v2** license in the same repo (link: https://github.com/google/styleguide/blob/gh-pages/LICENSE)

### CODE REVIEW

File: java/checkstyle/oilmod-checkstyle.xml [hunk 1]
```diff
@@ -3,6 +3,13 @@
         "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
         "https://checkstyle.org/dtds/configuration_1_3.dtd">
 
+<!-- 
+    This configuration has been adjusted from the original Google Checkstyle configuration,
+    which can be found at https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/google_checks.xml.
+
+    For more information, please refer Oilmod code style related article in Wiki.
```
Comment (keisernor) [inline java/checkstyle/oilmod-checkstyle.xml:10]: This is a public repository, the wiki article referred to is surely only internal?

**Review by keisernor — COMMENTED** [review]
The README has to be changed, in particular what is under item `License`, but this seems to be addressed separately.

File: java/intellij/intellij-java-oilmod-style.xml [hunk 2]
```diff
@@ -1,4 +1,12 @@
 <?xml version="1.0" encoding="UTF-8"?>
+
+<!-- 
+    This configuration has been adjusted from the original Google Code Style IntelliJ configuration,
+    which can be found at https://github.com/google/styleguide/blob/gh-pages/intellij-java-google-style.xml.
+
+    For more information, please refer Oilmod code style related article in Wiki.
```
[the remark this answers is not in the document]
Comment (dbelyaev) [reply java/intellij/intellij-java-oilmod-style.xml:7]: Removed reference in PR: https://github.com/equinor/oilmod1-code-style/pull/15

File: java/checkstyle/oilmod-checkstyle.xml (hunk 1 above)
Comment (dbelyaev) [reply java/checkstyle/oilmod-checkstyle.xml:10]: Removed reference in PR: https://github.com/equinor/oilmod1-code-style/pull/15

**Pull request merged**